### PR TITLE
Issue#466

### PR DESCRIPTION
**Pull request by HamidShojanazeri**
## Description

Fix the issue with inconsistent behavior of models between eager mode and Torchscript.

Fixes #(issue)
#466
## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

- [X] This change requires a documentation update

## testing

1. Use the setting in the for setup_config.json
[setup_config.log](https://github.com/pytorch/serve/files/4821681/setup_config.log)

3. Python Download_Transformer_models.py
4. torch-model-archiver --model-name DistillBERT_qa_Torchscript --version 1.0 --serialized-file
5. Transformer_model/traced_model.pt --handler ./Transformer_handler_generalized.py --extra-files "./setup_config.json"

6. mv DistillBERT_Torchscript.mar model_store

7. torchserve --start --model-store model_store --models my_tc=DistillBERT_qa_Torchscript.mar

8. curl -X POST http://127.0.0.1:8080/predictions/my_tc -T ./sample_text.txt

Repeat the above steps by modifying the first step and set the save_mode = "pretrained" in the setup_config.json
and for step 2:
torch-model-archiver --model-name DistillBERT_qa --version 1.0 --serialized-file Transformer_model/pytorch_model.bin --handler ./Transformer_handler_generalized.py --extra-files "Transformer_model/config.json,./setup_config.json"

- Logs
[run.log](https://github.com/pytorch/serve/files/4821684/run.log)

## Checklist:

- [X] Have you added tests that prove your fix is effective or that this feature works?
- [X] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: examples/Huggingface_Transformers/README.md [hunk 1]
```diff
@@ -58,6 +58,8 @@ In the setup_config.json :
 
 *model_name* : bert-base-uncased , roberta-base or other available pre-trained models.
 
+*fine_tuned* : name or path to the finetuned model if available, for example "distilbert-base-cased-distilled-squad", leave it blank if not available.
```
Comment (chauhang) [inline examples/Huggingface_Transformers/README.md:61]: There is already a model_name field. Can that not be used for fine tuned models name/path for the finetuned models as well? Please explain why we need a second field.

File: examples/Huggingface_Transformers/Download_Transformer_models.py [hunk 2]
```diff
@@ -5,25 +5,26 @@
...
 from transformers import (AutoModelForSequenceClassification, AutoTokenizer, AutoModelForQuestionAnswering,
  AutoModelForTokenClassification, AutoConfig)
+from transformers import set_seed
 """ This function, save the checkpoint, config file along with tokenizer config and vocab files
     of a transformer model of your choice.
 """
 print('Transformers version',transformers.__version__)
-
-def transformers_model_dowloader(mode,pretrained_model_name,num_labels,do_lower_case,max_length,torchscript):
+set_seed(1)
```
[the remark this answers is not in the document]
Comment (HamidShojanazeri) [reply examples/Huggingface_Transformers/Download_Transformer_models.py:13]: Thanks Geeta, without setting the seed in the Download script, it does not show consistent behavior/ reproducible results  between eager mode and torchscript, when we are not using fine-tuned models. I believe this can be result of using torch.jit.trace as it may introduce some randomness.

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py [hunk 3]
```diff
@@ -5,11 +5,11 @@
 import ast
 import torch
 from transformers import AutoModelForSequenceClassification, AutoTokenizer, AutoModelForQuestionAnswering,AutoModelForTokenClassification
-
+from transformers import set_seed
 from ts.torch_handler.base_handler import BaseHandler
 
 logger = logging.getLogger(__name__)
-
+set_seed(1)
```
[the remark this answers is not in the document]
Comment (HamidShojanazeri) [reply examples/Huggingface_Transformers/Transformer_handler_generalized.py]: it's been removed.

File: examples/Huggingface_Transformers/README.md (hunk 1 above)
Comment (HamidShojanazeri) [reply examples/Huggingface_Transformers/README.md:61]: In the current setting , when we do not pass the vocab.txt as the artifacts, tokenizer uses the pretrained version as being passed by Model_name. In that case fine_tuned version can not be used for the tokenizer.
